docs: child-to-parent-messaging (#2154)
OffchainLabs/arbitrum-docs: arbitrum-docs/how-arbitrum-works/11-l2-to-l1-messaging.mdx

The following diagram depicts the process that funds follow during a withdrawal operation.

<ImageZoom
  src="/img/dapps-withdrawing-ether.svg"
  alt="Process funds follow during withdrawal operation"
  className="img-600px"
/>

## `ERC-20` token withdrawal

Arbitrum has a canonical <a data-quicklook-from="bridge">bridge</a> design and architecture, which we explain in detail in the [Token bridging](/how-arbitrum-works/10-l1-to-l2-messaging.mdx#token-bridging) section of Bridging from a parent chain to a child chain article. This section will explain how the Arbitrum canonical bridge works on the child-to-parent chain token bridging.

1. Initiation of a child-to-parent chain transfer occurs via the `L2GatewayRouter` contract on the child chain.
2. The token's gateway contract (`L2ArbitrumGateway`) on the child chain gets called.

> 3. Which finally communicates to its corresponding gateway contract on the parent chain using the `L1ArbitrumGateway` contract.

Reviewer: ```suggestion
3. `L2ArbitrumGateway` finally communicates to its corresponding gateway contract on the parent chain using the `L1ArbitrumGateway` contract.
```

<ImageZoom
  src="/img/dapps-bridge_withdrawals.png"
  alt="Withdrawal process using the gateway"
  className="img-600px"
/>

> For any given gateway pairing, we require initiation of calls through the corresponding router (`L2GatewayRouter`) and that the gateways conform to the [TokenGateway] interfaces; the `TokenGateway` interfaces should be flexible and extensible enough to support any bridging functionality a particular token may require.

Reviewer: ```suggestion
For any given gateway pairing, calls have to be initiated through the corresponding router (`L2GatewayRouter`), and the gateways must conform to the `TokenGateway` interfaces; the `TokenGateway` interfaces should be flexible and extensible enough to support any bridging functionality a particular token may require.
```

> Token withdrawals are the most common usage of child-to-parent chain messaging. The standard withdrawal process follows (for standard gateway):

Reviewer: ```suggestion
Token withdrawals are the most common usage of child-to-parent chain messaging. The standard withdrawal happens as follows (for standard gateway):
```

> 1. On the child chain, a user calls `L2GatewayRouter.outBoundTransfer`, which calls `outBoundTransfer` on arbSomeERC20Token's gateway (i.e., L2ERC20Gateway).

Reviewer: ```suggestion
1. On the child chain, a user calls `L2GatewayRouter.outBoundTransfer`, which calls `outBoundTransfer` on `arbSomeERC20Token`'s gateway (i.e., L2ERC20Gateway).
```

> content_type: get-started

Reviewer: ```suggestion
content_type: concept
```

> content_type: get-started

Reviewer: ```suggestion
content_type: concept
```

| Confirmed and executable on the parent chain | If no <a data-quicklook-from="fraud-proof">fraud proof</a> is submitted, the assertion is confirmed, and the message is available for execution in the outbox. |

## Withdrawing Ether

Withdrawing Ether can be done using the [ArbSys precompile](/build-decentralized-apps/precompiles/02-reference.mdx#arbsys)'s `withdrawEth` method:

```solidity
ArbSys(100).withdrawEth{ value: 2300000 }(destAddress)
```

Upon withdrawing, the Ether balance is burnt on the child chain side and will later be made available on the parent chain side.

`ArbSys.withdrawEth` is a convenience function equivalent to calling `ArbSys.sendTxToL1` with empty `calldataForL1`. Like any other `sendTxToL1` call, it will require an additional call to `Outbox.executeTransaction` on the parent chain after the dispute period elapses for the user to finalize claiming their funds on the parent chain. Once the withdraw executes (removed from the outbox), the user's Ether balance will receive credit on the parent chain.

The following diagram depicts the process that funds follow during a withdrawal operation.

> <ImageZoom

Author: Pushed updated image
